runtime/caml/platform.h

```diff
@@ -414,14 +417,16 @@ Caml_inline unsigned caml_plat_spin_step(unsigned spins,
   }
 }
 
-#define SPIN_WAIT_BACK_OFF(max_spins)                                   \
-  unsigned CAML_GENSYM(spins) = 0;                                      \
-  unsigned CAML_GENSYM(max_spins) = (max_spins);                        \
-  static const struct caml_plat_srcloc CAML_GENSYM(loc) = {             \
-    __FILE__, __LINE__, __func__                                        \
-  };                                                                    \
-  for (; 1; CAML_GENSYM(spins) = caml_plat_spin_step(                   \
-         CAML_GENSYM(spins), CAML_GENSYM(max_spins), &CAML_GENSYM(loc)))
+#define SPIN_WAIT_BACK_OFF_(spins, max_spins, N, loc)           \
+  static const struct caml_plat_srcloc loc =                    \
+    { __FILE__, __LINE__, __func__ };                           \
+  for (unsigned int spins = 0, max_spins = (N);                 \
+       true;                                                    \
+       spins = caml_plat_spin_step(spins, max_spins, &loc))
+
+#define SPIN_WAIT_BACK_OFF(N)                                           \
+  SPIN_WAIT_BACK_OFF_(CAML_GENSYM(spins), CAML_GENSYM(max_spins), (N),  \
+                      CAML_GENSYM(loc))
```

Comment:
I would prefer if the symbols went before together, and the real parameters came after. In fact at the declaration site there could be a clear separation:

```c
#define SPIN_WAIT_BACK_OFF_(/* symbols */ spins, max_spins, loc,               \
                            /* params */ N)                                    \
```